config/environments/test.rb

```diff
@@ -11,10 +11,10 @@
   # Turn false under Spring and add config.action_view.cache_template_loading = true.
   config.cache_classes = true
 
-  # Eager loading loads your whole application. When running a single test locally,
-  # this probably isn't necessary. It's a good idea to do in a continuous integration
-  # system, or in some way before deploying your code.
-  config.eager_load = ENV["CI"].present?
+  # Do not eager load code on boot. This avoids loading your whole application
+  # just for the purpose of running a single test. If you are using a tool that
+  # preloads Rails for running tests, you may have to set it to true.
+  config.eager_load = false
```

Comment:
@ddzz  , the root cause makes sense, but we shouldn't disable eager loading in CI, as that helps catch any loading issues that we might not run into locally. I have been trying a few strategies to work around this, but I think our best bet is to explicitly load the necessary fixtures in CircleCI before we run the cucumber tests, with a comment explaining why. This is what I'm thinking:
```yaml
#.circleci/config.yml

  cucumber:
...
    steps:
....
      - prepare_database

      # In CI, the application is eager loaded. This ensures the db contains the necessary
      # records before the application is loaded.
      - run:
          name: Load Database Fixtures
          command: RAILS_ENV=test rails db:fixtures:load FIXTURES_PATH='./spec/fixtures' FIXTURES=languages,users....#(see the list in features/support/hooks.rb)
```
I'm not in love with that solution either, so I'm open to ideas.

Reply:
We could load the fixtures through `db/seeds.rb`. With that approach we don't need an extra step in CircleCI.